### ISSUE TITLE
Units other than meters and degrees

### ISSUE THREAD

**Issue by lf73** [developmentseed/morecantile#85]
The referenced specification only considers Meters and Degrees.  Would it be possible to consider other units?  In the United States were have the State Plane Coordinate System with either defined units in "US Survey Foot" or "International Foot".

US Survey Foot = 0.3048006096
International Foot = 0.3048

I hard-coded the else statement to the US Survey Foot for EPSG:2276 data that I am working with just as a test. And I am very happy to have found success in properly tiling locally referenced imagery!

I don't know Python, but I could take a crack at adding a case statement for the defined units if there is no objection. 

https://github.com/developmentseed/morecantile/blob/17020c555d4f09e3027cab755cf0f74744106774/morecantile/utils.py#L55

From note g in http://docs.opengeospatial.org/is/17-083r2/17-083r2.html#table_2:
        If the CRS uses meters as units of measure for the horizontal dimensions,
        then metersPerUnit=1; if it has degrees, then metersPerUnit=2pa/360
        (a is the Earth maximum radius of the ellipsoid).

**Comment by vincentsarago** [comment]
Thanks for opening the issue @lf73 

I've opened a PR over https://github.com/developmentseed/morecantile/pull/86, it would be nice if you could confirm that this will fix your issue 🙏

**Comment by lf73** [comment]
@vincentsarago  Thank you very much for the elegant fix. I confirmed this solves the issue with all of the units. I confirmed with 2276, US Survey Foot; 2222, Foot; 4326, Degree; and 32631 Metre.  I am not aware of any other units in use.

Appreciate your help. TiTiler is working beautifully with the correct unit in place and hopefully this opens an opportunity for others to utilize. I am very happy not having to do projections and being able to keep native to our area. 

By the way, the United States is making progress to international units! :-)  Just this year we eliminated the State Planes Coordinate Systems from using the US Survey Foot!! Only to go to the International foot!  Maybe in another 20 years, we will progress to the Metre, albeit we'll probably have to spell as Meter.....
